File: 백준/Silver/8979. 올림픽/올림픽.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int N, K;
    vector<vector<int>> medals;

    cin >> N >> K;
    
    int nation, gold, silver, copper;
    vector<int> target;
    for (int i = 0; i < N; i++) {
        cin >> nation >> gold >> silver >> copper;
        medals.push_back({gold, silver, copper});
        if (nation == K) {
            target = {gold, silver, copper};
        }
    }

    int rank = 1;
    for (auto medal : medals) {
        if (medal > target) {
            rank++;
        }
    }
    
    cout << rank;
    
    return 0;
}

File: 백준/Silver/8979. 올림픽/올림픽.py
import sys
from collections import defaultdict

input = sys.stdin.readline

N, K = map(int, input().split())
medals = defaultdict(list)
for _ in range(N):
    nation, gold, silver, copper = map(int, input().split())
    medals[(gold, silver, copper)].append(nation)
sorted_by_medals = sorted(medals, reverse=True)

rank = 0
for medal in sorted_by_medals:
    rank = len(medals[medal])
    if K in medals[medal]:
        break

print(rank)
